docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,447 @@
+
+It does not require having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connector-plugins ls
+PLUGIN                          
+postgres@v0.2.0                 
+conduit-connector-http@0.1.0
+```
+
+### `conduit connector-plugins PLUGIN`
+
+This command will describe the configuration for that connector plugin.
+
+It does not require having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connnector-plugins describe PLUGIN
+
+EXAMPLE:
+
+$ conduit connector-plugins describe conduit-connector-http@0.1.0
+NAME   DESCRIPTION                       REQUIRED  DEFAULT VALUE  EXAMPLE
+url    HTTP URL to send requests to.     true                     https://...
+```
+
+### `conduit processors ls`
+
+This command will list all the available processors.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit processors ls
+NAME          TYPE      PIPELINE
+avro.decode   builtin   my-pipeline
+avro.encode    builtin   my-pipeline
+base64.decode  builtin   my-other-pipeline
+```
+
+### `conduit processors describe ID`
+
+This command will describe the existing running processor.
+
+It requires having Conduit previously running.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| ID  |  processor ID | Yes | |
+
+#### `--help`
+
+```bash
+$ conduit processors describe ID
+
+EXAMPLE:
+
+$ conduit processors describe my-processor
+NAME            PLUGIN            PIPELINE
+my-processor    base64.decode     my-pipeline
+```
+
+### `conduit processor-plugins ls`
+
+This command will list all the available processors.
+
+It does not require having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit processor-plugins ls
+NAME
+avro.decode
+avro.encode
+base64.decode
+```
+
+### `conduit processor-plugins describe PLUGIN`
+
+This command will describe the processor plugin configuration available.
+
+It does not require having Conduit previously running.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| PLUGIN  |  processor plugin name | Yes | |
+
+#### `--help`
+
+```bash
+$ conduit processor-plugins describe PLUGIN
+
+EXAMPLE:
+
+$ conduit processor-plugins describe avro.decode
+# auth.basic.password
+type: string
+description: This option is required if auth.basic.username contains a value. If both auth.basic.username and auth.basic.password are empty basic authentication is disabled.
+
+# auth.basic.username
+type: string
+...
+```
+
+### `conduit doctor`
+
+This will check whether there’s a more up to date version of Conduit.
+
+Also will check iff some connectors or processors could be updated.
+
+It requires having Conduit previously running.
```

Comment:
Why does this require a running instance of Conduit? 🤔

Also, IMO we could do without this command at first, I think it increases the scope.

Reply:
@lovromazgon yeah, you're right. This sounds like a nice to have.